src/hardware/serialport/softmodem.cpp

```diff
@@ -288,6 +288,18 @@ void CSerialModem::SendRes(const ResTypes response) {
 }
 
 bool CSerialModem::Dial(const char * host) {
+	// Close the server socket before dialing
+	waitingclientsocket.reset(nullptr);
+
+	if (serversocket) {
+		waitingclientsocket.reset(serversocket->Accept());
+		while (waitingclientsocket) {
+			waitingclientsocket.reset(serversocket->Accept());
+		}
+
+		serversocket.reset(nullptr);
+	}
```

Comment:
The function `CSerialModem::EnterIdleState()` contains the same code that discards all the server connections:

```c++
waitingclientsocket.reset(nullptr);
if (serversocket) {
	waitingclientsocket.reset(serversocket->Accept());
	while (waitingclientsocket) {
		waitingclientsocket.reset(serversocket->Accept());
	}
	...
```

Suggest de-duping to something like:

```c++
void CSerialModem::CloseServerSocket() {
    waitingclientsocket.reset();

    if (serversocket) {
        while (auto accepted = serversocket->Accept()) {
            delete accepted;
        }
        serversocket.reset();
    }
}
```

Reply:
Seem reasonable to reuse that code/logic and not copy/paste it into two places, yeah.

Reply:
```
..\src\hardware\serialport\misc_util.cpp(111,25): warning : call to pure virtual member function 'Accept' has undefined behavior; overrides of 'Accept' in subclasses are not available in the destructor of 'NETServerSocket' [-Wcall-to-pure-virtual-from-ctor-dtor]
```
Seem that this won't be that simple.

Reply:
Yeah, like @interloper98 said.